pyorthanc/_resources/patient.py

```diff
@@ -242,23 +253,136 @@ def anonymize(self, remove: List = None, replace: Dict = None, keep: List = None
             List of tag to keep unchanged
         force
             Some tags can't be changed without forcing it (e.g. PatientID) for security reason
+        keep_private_tags
+            If True, keep the private tags from the DICOM instances.
+        keep_source
+            If False, instructs Orthanc to the remove original resources.
+            By default, the original resources are kept in Orthanc.
+        priority
+            In asynchronous mode, the priority of the job. The lower the value, the higher the priority.
+        permissive
+            If True, ignore errors during the individual steps of the job.
+        dicom_version
+            Version of the DICOM standard to be used for anonymization.
+            Check out configuration option DeidentifyLogsDicomVersion for possible values.
 
         Returns
         -------
         Patient
             A New anonymous patient.
+
+        Examples
+        --------
+        ```python
+        new_patient = patient.anonymize()
+
+        new_patient_with_specific_patient_id = patient.anonymize(
+            keep=['PatientName'],
+            replace={'PatientID': 'TheNewPatientID'},
+            force=True
+        )
         """
         remove = [] if remove is None else remove
         replace = {} if replace is None else replace
         keep = [] if keep is None else keep
 
-        anonymous_patient = self.client.post_patients_id_anonymize(
-            self.id_,
-            json={'Remove': remove, 'Replace': replace, 'Keep': keep, 'Force': force}
-        )
+        data = {
+            'Asynchronous': False,
+            'Remove': remove,
+            'Replace': replace,
+            'Keep': keep,
+            'Force': force,
+            'KeepPrivateTags': keep_private_tags,
+            'KeepSource': keep_source,
+            'Priority': priority,
+            'Permissive': permissive,
+        }
+        if dicom_version is not None:
+            data['DicomVersion'] = dicom_version
+
+        try:
+            anonymous_patient = self.client.post_patients_id_anonymize(self.id_, data)
+        except ReadTimeout:
+            raise ReadTimeout('Patient is too long to process. Use `.anonymize_as_job` or increase client.timeout.')
 
         return Patient(anonymous_patient['ID'], self.client)
 
+    def anonymize_as_job(self, remove: List = None, replace: Dict = None, keep: List = None,
+                         force: bool = False, keep_private_tags: bool = False,
+                         keep_source: bool = True, priority: int = 0, permissive: bool = False,
+                         dicom_version: str = None) -> Job:
+        """Anonymize patient and return a job
+
+        If no error has been raise, then it creates a new anonymous patient.
```

Comment:
raised ?

Reply:
You're right! In the "job" version, we should not have an error if the job failed